solution_template/jenkins/MainTemplate.json

```diff
@@ -0,0 +1,331 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "_artifactsLocation": {
+      "metadata": {
+        "description": "Artifacts location"
+      },
+      "type": "string"
+    },
+    "_artifactsLocationSasToken": {
+      "metadata": {
+        "description": "Artifacts Sas token"
+      },
+      "type": "string"
+    },
+    "adminPassword": {
+      "metadata": {
+        "description": "Password for the Virtual Machine. Will be used only is authenticationType is 'password'"
+      },
+      "type": "securestring"
+    },
+    "adminSSHPublicKey": {
+      "metadata": {
+        "description": "Public SSH key for the Virtual Machine. Will be used only is authenticationType is 'sshPublicKey'"
+      },
+      "type": "securestring"
+    },
+    "adminUserName": {
+      "metadata": {
+        "description": "User name for the Virtual Machine"
+      },
+      "type": "string"
+    },
+    "authenticationType": {
+      "metadata": {
+        "description": "Authentication type (can be 'password' or 'sshPublicKey')"
+      },
+      "type": "string"
+    },
+    "dnsPrefix": {
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Jenkins Virtual Machine."
+      },
+      "type": "string"
+    },
+    "publicIPName": {
+      "metadata": {
+        "description": "Public IP name"
+      },
+      "type": "string"
+    },
+    "publicIPNewOrExisting": {
+      "metadata": {
+        "description": "Should be 'new' if the public IP has to be created"
+      },
+      "type": "string"
+    },
+    "storageAccountName": {
+      "metadata": {
+        "description": "Storage Account Name"
+      },
+      "type": "string"
+    },
+    "storageAccountNewOrExisting": {
+      "metadata": {
+        "description": "Should be 'new' or 'existing'"
+      },
+      "type": "string"
+    },
+    "storageAccountType": {
+      "allowedValues": [
```

Comment:
Why validate the values at this point? Shouldn't it be done in the UI? If this provides value, then consider adding allowedValues for other parameters like authenticationType and storageAccountNewOrExisting